bindings/go/oci/repository.go

```diff
@@ -178,6 +180,9 @@ func (repo *Repository) GetComponentVersion(ctx context.Context, component, vers
 	}
 
 	desc, _, _, err = getDescriptorFromStore(ctx, store, reference)
+	if errors.Is(err, errdef.ErrNotFound) {
+		return desc, errors.Join(fmt.Errorf("component version %s/%s not found: %w", component, version, componentrepository.ErrNotFound), err)
```

Comment:
```suggestion
		return desc, errors.Join(componentrepository.ErrNotFound, fmt.Errorf("component version %s/%s not found: %w", component, version, err))
```
i think you should switch this around as the first error will be the first one shown in the join